Java 11 EA Build: Compile Errors: ambiguous reference both <T>assertThat(T) and method assertThat(java.lang.CharSequence) match

Just for your information.

When running `mvn clean test` using:

```
java version "11-ea" 2018-09-25
Java(TM) SE Runtime Environment 18.9 (build 11-ea+10)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11-ea+10, mixed mode)
```

I get these compile errors:

```
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/AutoCloseableBDDSoftAssertionsTest.java:[62,13] reference to then is ambiguous
  both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/SoftAssertionsTest.java:[219,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/BDDSoftAssertionsTest.java:[216,13] reference to then is ambiguous
  both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/perf/SoftAssertionsPerfTest.java:[192,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/AutoCloseableSoftAssertionsTest.java:[71,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[35,61] reference to assertThat is ambiguous
  both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[41,61] reference to assertThat is ambiguous
  both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
```

For info, same error in my environment.

**Java version**
``` 
~/prog/assertj/assertj-core-3.x ∙ java -version
openjdk version "11-ea" 2018-09-25
OpenJDK Runtime Environment 18.9 (build 11-ea+13)
OpenJDK 64-Bit Server VM 18.9 (build 11-ea+13, mixed mode)
``` 

**Maven version**
```
~/prog/assertj/assertj-core-3.x ∙ mvn -version
Apache Maven 3.5.3 (3383c37e1f9e9b3bc3df5050c29c8aff9f295297; 2018-02-25T08:49:05+13:00)
Maven home: /home/joel/prog/apache-maven-3.5.3
Java version: 11-ea, vendor: Oracle Corporation
Java home: /home/joel/prog/java/jdk-11
Default locale: en_NZ, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-124-generic", arch: "amd64", family: "unix"
``` 

Output of `mvn clean test`:
```
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] ----------------------< org.assertj:assertj-core >----------------------
[INFO] Building AssertJ fluent assertions 3.11.0-SNAPSHOT
[INFO] -------------------------------[ bundle ]-------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ assertj-core ---
[INFO] Deleting /home/joel/prog/assertj/assertj-core-3.x/target
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ assertj-core ---
[INFO] Updating license headers...
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.1:prepare-agent (prepare-agent) @ assertj-core ---
[INFO] argLine set to -javaagent:/home/joel/.m2/repository/org/jacoco/org.jacoco.agent/0.8.1/org.jacoco.agent-0.8.1-runtime.jar=destfile=/home/joel/prog/assertj/assertj-core-3.x/target/jacoco.exec,excludes=**/*hamcrest*/** --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED
 --add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.math=ALL-UNNAMED
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ assertj-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/joel/prog/assertj/assertj-core-3.x/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:compile (default-compile) @ assertj-core ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 579 source files to /home/joel/prog/assertj/assertj-core-3.x/target/classes
[INFO] /home/joel/prog/assertj/assertj-core-3.x/src/main/java/org/assertj/core/api/AbstractDoubleAssert.java: Some input files use or override a deprecated API.
[INFO] /home/joel/prog/assertj/assertj-core-3.x/src/main/java/org/assertj/core/api/AbstractDoubleAssert.java: Recompile with -Xlint:deprecation for details.
[INFO] /home/joel/prog/assertj/assertj-core-3.x/src/main/java/org/assertj/core/api/AbstractIterableAssert.java: Some input files use unchecked or unsafe operations.
[INFO] /home/joel/prog/assertj/assertj-core-3.x/src/main/java/org/assertj/core/api/AbstractIterableAssert.java: Recompile with -Xlint:unchecked for details.
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ assertj-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 17 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:testCompile (default-testCompile) @ assertj-core ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2730 source files to /home/joel/prog/assertj/assertj-core-3.x/target/test-classes
[INFO] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/FloatAssertBaseTest.java: Some input files use or override a deprecated API.
[INFO] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/FloatAssertBaseTest.java: Recompile with -Xlint:deprecation for details.
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/AutoCloseableBDDSoftAssertionsTest.java:[62,13] reference to then is ambiguous
  both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/perf/SoftAssertionsPerfTest.java:[214,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/SoftAssertionsTest.java:[219,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[35,61] reference to assertThat is ambiguous
  both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[41,61] reference to assertThat is ambiguous
  both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/BDDSoftAssertionsTest.java:[216,13] reference to then is ambiguous
  both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/AutoCloseableSoftAssertionsTest.java:[71,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[INFO] 7 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 48.899 s
[INFO] Finished at: 2018-05-19T11:24:59+12:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:testCompile (default-testCompile) on project assertj-core: Compilation failure: Compilation failure: 
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/AutoCloseableBDDSoftAssertionsTest.java:[62,13] reference to then is ambiguous
[ERROR]   both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/perf/SoftAssertionsPerfTest.java:[214,13] reference to assertThat is ambiguous
[ERROR]   both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/SoftAssertionsTest.java:[219,13] reference to assertThat is ambiguous
[ERROR]   both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[35,61] reference to assertThat is ambiguous
[ERROR]   both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[41,61] reference to assertThat is ambiguous
[ERROR]   both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/BDDSoftAssertionsTest.java:[216,13] reference to then is ambiguous
[ERROR]   both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /home/joel/prog/assertj/assertj-core-3.x/src/test/java/org/assertj/core/api/AutoCloseableSoftAssertionsTest.java:[71,13] reference to assertThat is ambiguous
[ERROR]   both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

--- a/src/main/java/org/assertj/core/api/Assertions.java
+++ b/src/main/java/org/assertj/core/api/Assertions.java
@@ -2566,7 +2566,7 @@ public class Assertions {
   }
 
   /**
-   * Creates a new instance of <code>{@link CharSequenceAssert}from a {@link StringBuilder}</code>.
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}.
    *
    * @param actual the actual value.
    * @return the created assertion object.
@@ -2574,10 +2574,10 @@ public class Assertions {
    */
   @CheckReturnValue
   public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuilder actual) {
-    return new CharSequenceAssert(actual);
+    return AssertionsForClassTypes.assertThat(actual);
   }
   /**
-   * Creates a new instance of <code>{@link CharSequenceAssert}from a {@link StringBuffer}</code>.
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuffer}.
    *
    * @param actual the actual value.
    * @return the created assertion object.
@@ -2585,7 +2585,7 @@ public class Assertions {
    */
   @CheckReturnValue
   public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuffer actual) {
-    return new CharSequenceAssert(actual);
+    return AssertionsForClassTypes.assertThat(actual);
   }
 
   /**

--- a/src/main/java/org/assertj/core/api/Assertions.java
+++ b/src/main/java/org/assertj/core/api/Assertions.java
@@ -2566,6 +2566,29 @@ public class Assertions {
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}from a {@link StringBuilder}</code>.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuilder actual) {
+    return new CharSequenceAssert(actual);
+  }
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}from a {@link StringBuffer}</code>.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuffer actual) {
+    return new CharSequenceAssert(actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link CharSequenceAssert}from a {@link String}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/AssertionsForClassTypes.java
+++ b/src/main/java/org/assertj/core/api/AssertionsForClassTypes.java
@@ -488,6 +488,30 @@ public class AssertionsForClassTypes {
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}..
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuilder actual) {
+    return new CharSequenceAssert(actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuffer actual) {
+    return new CharSequenceAssert(actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link StringAssert}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/AssertionsForClassTypes.java
+++ b/src/main/java/org/assertj/core/api/AssertionsForClassTypes.java
@@ -488,7 +488,7 @@ public class AssertionsForClassTypes {
   }
 
   /**
-   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}..
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}.
    *
    * @param actual the actual value.
    * @return the created assertion object.

--- a/src/main/java/org/assertj/core/api/Assumptions.java
+++ b/src/main/java/org/assertj/core/api/Assumptions.java
@@ -309,6 +309,31 @@ public class Assumptions {
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> assumption from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assumeThat(StringBuilder actual) {
+    return asAssumption(CharSequenceAssert.class, CharSequence.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> assumption from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assumeThat(StringBuffer actual) {
+    return asAssumption(CharSequenceAssert.class, CharSequence.class, actual);
+  }
+
+
+  /**
    * Creates a new instance of <code>{@link ShortAssert}</code> assumption.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/BDDAssertions.java
+++ b/src/main/java/org/assertj/core/api/BDDAssertions.java
@@ -801,6 +801,30 @@ public class BDDAssertions extends Assertions {
   }
 
   /**
+   * Creates a new instance of <code>{@link org.assertj.core.api.CharSequenceAssert}</code> from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> then(StringBuilder actual) {
+    return assertThat(actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link org.assertj.core.api.CharSequenceAssert}</code> from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> then(StringBuffer actual) {
+    return assertThat(actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link org.assertj.core.api.StringAssert}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/Java6AbstractBDDSoftAssertions.java
+++ b/src/main/java/org/assertj/core/api/Java6AbstractBDDSoftAssertions.java
@@ -488,6 +488,30 @@ public class Java6AbstractBDDSoftAssertions extends AbstractSoftAssertions {
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public CharSequenceAssert then(StringBuilder actual) {
+    return proxy(CharSequenceAssert.class, CharSequence.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public CharSequenceAssert then(StringBuffer actual) {
+    return proxy(CharSequenceAssert.class, CharSequence.class, actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link StringAssert}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/Java6AbstractStandardSoftAssertions.java
+++ b/src/main/java/org/assertj/core/api/Java6AbstractStandardSoftAssertions.java
@@ -490,6 +490,30 @@ public class Java6AbstractStandardSoftAssertions extends AbstractSoftAssertions 
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public CharSequenceAssert assertThat(StringBuilder actual) {
+    return proxy(CharSequenceAssert.class, CharSequence.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public CharSequenceAssert assertThat(StringBuffer actual) {
+    return proxy(CharSequenceAssert.class, CharSequence.class, actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link StringAssert}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/Java6Assertions.java
+++ b/src/main/java/org/assertj/core/api/Java6Assertions.java
@@ -925,6 +925,30 @@ public class Java6Assertions {
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuilder actual) {
+    return new CharSequenceAssert(actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuffer actual) {
+    return new CharSequenceAssert(actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link StringAssert}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/Java6BDDAssertions.java
+++ b/src/main/java/org/assertj/core/api/Java6BDDAssertions.java
@@ -796,6 +796,30 @@ public class Java6BDDAssertions {
   }
 
   /**
+   * Creates a new instance of <code>{@link org.assertj.core.api.CharSequenceAssert}</code> from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> then(StringBuilder actual) {
+    return assertThat(actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link org.assertj.core.api.CharSequenceAssert}</code> from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> then(StringBuffer actual) {
+    return assertThat(actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link org.assertj.core.api.StringAssert}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/WithAssertions.java
+++ b/src/main/java/org/assertj/core/api/WithAssertions.java
@@ -565,6 +565,30 @@ public interface WithAssertions {
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  default AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(final StringBuilder actual) {
+    return Assertions.assertThat(actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  default AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(final StringBuffer actual) {
+    return Assertions.assertThat(actual);
+  }
+
+  /**
    * Creates a new instance of <code>{@link ShortArrayAssert}</code>.
    *
    * @param actual the actual value.

--- a/src/main/java/org/assertj/core/api/WithAssumptions.java
+++ b/src/main/java/org/assertj/core/api/WithAssumptions.java
@@ -372,6 +372,29 @@ public interface WithAssumptions {
   }
 
   /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> assumption from a {@link StringBuilder}.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  default AbstractCharSequenceAssert<?, ? extends CharSequence> assumeThat(final StringBuilder actual) {
+    return Assumptions.assumeThat(actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}</code> assumption from a {@link StringBuffer}.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  default AbstractCharSequenceAssert<?, ? extends CharSequence> assumeThat(final StringBuffer actual) {
+    return Assumptions.assumeThat(actual);
+  }
+  /**
    * Creates a new instance of <code>{@link ShortArrayAssert}</code> assumption.
    *
    * @param actual the actual value.

--- a/src/test/java/org/assertj/core/api/Assertions_assertThat_with_StringBuffer_Test.java
+++ b/src/test/java/org/assertj/core/api/Assertions_assertThat_with_StringBuffer_Test.java
@@ -1,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for <code>{@link Assertions#assertThat(StringBuffer)}</code>.
+ */
+public class Assertions_assertThat_with_StringBuffer_Test {
+
+  @Test
+  public void should_create_Assert() {
+    AbstractCharSequenceAssert<?, ?> assertions = Assertions.assertThat(new StringBuffer("Yoda"));
+    assertThat(assertions).isNotNull();
+  }
+
+  @Test
+  public void should_pass_actual() {
+    StringBuffer actual = new StringBuffer("Yoda");
+    AbstractCharSequenceAssert<?, ?> assertions = Assertions.assertThat(actual);
+    assertThat(assertions.actual).isSameAs(actual);
+  }
+}

--- a/src/test/java/org/assertj/core/api/Assertions_assertThat_with_StringBuilder_Test.java
+++ b/src/test/java/org/assertj/core/api/Assertions_assertThat_with_StringBuilder_Test.java
@@ -1,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for <code>{@link Assertions#assertThat(StringBuilder)}</code>.
+ */
+public class Assertions_assertThat_with_StringBuilder_Test {
+
+  @Test
+  public void should_create_Assert() {
+    AbstractCharSequenceAssert<?, ?> assertions = Assertions.assertThat(new StringBuilder("Yoda"));
+    assertThat(assertions).isNotNull();
+  }
+
+  @Test
+  public void should_pass_actual() {
+    StringBuilder actual = new StringBuilder("Yoda");
+    AbstractCharSequenceAssert<?, ?> assertions = Assertions.assertThat(actual);
+    assertThat(assertions.actual).isSameAs(actual);
+  }
+}

